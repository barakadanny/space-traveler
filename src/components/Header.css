header {
    background-color: var(--white-color);
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 1em 1em;
}

nav {
    display: flex;
    gap: 0.5rem;
    align-items: center;
    cursor: pointer;
}

.logo {
    width: 30px;
    cursor: pointer;
}

nav h1 {
    color: var(--dark-bold-color);
    font-weight: 700;
    letter-spacing: 1px;
}

header ul {
    display: flex;
    gap: 1rem;
}

header span {
    color: var(--azure-color);
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    padding: 8px;
    border-radius: 50%;
    border: 1px solid #e4dede;
}

/* .nav-menu {
    color: red;
} */

/* Desktop View */
@media screen and (min-width: 768px) {
    header {
        padding: 1em 5rem;
    }
}