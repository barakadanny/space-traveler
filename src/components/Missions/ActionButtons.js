import PropTypes from 'prop-types';

const ActionButton = ({ action }) => (<button type="button">{action}</button>);

ActionButton.propTypes = {
  action: PropTypes.string.isRequired,
};

export default ActionButton;
