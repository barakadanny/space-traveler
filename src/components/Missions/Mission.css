.missions-container {
    font-family: Arial, Helvetica, sans-serif;
    border-collapse: collapse;
    background-color: #fff;
    width: 100%;
  }
  
  .missions-container td, .missions-container th {
    border: 1px solid #ddd;
    padding: 8px;
  }
  
  .missions-container tr:nth-child(even){background-color: #f2f2f2;}
  
.missions-container tr:hover {background-color: #ddd;}
  
  .missions-container th {
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: left;
    background-color: #fff;
    font-size: 20px;
    
  }

  .reg-status button {
    background-color: rgb(85, 78, 78);
    color: #fff;
    border-radius: 5px;
    width: 150px;
    padding: 5px;
  }