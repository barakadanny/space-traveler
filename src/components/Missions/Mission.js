import React from "react";
import ActionButton from "./ActionButtons";
import { useDispatch } from "react-redux";

function Mission(props) {
  const { title, description, index } = props;

  const dispatch = useDispatch();

  //   console.log(title);
  return (
    <tr>
      <td>{title}</td>
      <td>{description}</td>
      <td className="reg-status">
        <ActionButton action="NOT A MEMBER" />
      </td>
      <td>
        <ActionButton action="Join Mission" />
      </td>
    </tr>
  );
}

export default Mission;
