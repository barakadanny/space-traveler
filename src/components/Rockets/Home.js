import React from 'react';
import RockectsList from './RocketsList';

const Home = () => (
  <main>
    <RockectsList />
  </main>
);

export default Home;
