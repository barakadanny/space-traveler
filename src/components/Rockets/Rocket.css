.single-rocket {
  background-color: var(--white-color);
  display: flex;
  flex-direction: column;
  gap: 1rem;
  padding: 1rem;
  align-items: center;
}

.img-container {
  max-width: 200px;
  position: relative;
}

.img-container img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.rocket-info h3,
p {
  margin-bottom: 0.5rem;
}

/* Desktop view */
@media screen and (min-width: 768px) {
  .single-rocket {
    flex-direction: row;
  }
}
