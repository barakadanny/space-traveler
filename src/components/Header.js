import React from "react";
// import { Link } from "react-router-dom";
import "./Header.css";
// import { FaUser } from "react-icons/fa";

function Header() {
  //   const linkStyle = {
  //     textDecoration: "none",
  //     color: "#121212",
  //   };
  return (
    <header>
      <nav>
        <img className="logo" src={process.env.PUBLIC_URL + "planet.png"} />
        <h1>Space Traveler's Hub</h1>
      </nav>
      <ul className="nav-menu">
        <li>
          {/* <Link style={linkStyle} to="/">
              BOOKS
            </Link> */}
          Rockets
        </li>
        <li>
          {/* <Link style={linkStyle} to="/Categories">
              CATEGORIES
            </Link> */}
          Missions
        </li>
      </ul>
    </header>
  );
}

export default Header;
