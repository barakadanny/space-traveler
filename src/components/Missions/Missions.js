import React from "react";
import "./Mission.css";
import Mission from "./Mission";
import { useSelector } from "react-redux";

function Missions() {
  const missions = useSelector((state) => state.mission);

  return (
    <table className="missions-container">
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th>-</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <Mission
            key={mission.mission_id}
            index={mission.mission_id}
            title={mission.mission_name}
            description={mission.description}
          />
          // <Mission />
        ))}
      </tbody>

      {/* <tr>
        <td>Telstar</td>
        <td>
          The Missions section displays a list of current missions along with
          their brief description and participation status. There is also a
          button next to each mission that allows users to join the selected
          mission or leave the mission the user joined earlier.
        </td>
        <td>
          <ActionButton action="ACTIVE MEMBER" />
        </td>
        <td>
          <ActionButton action="Leave Mission" />
        </td>
      </tr> */}
    </table>
  );
}

export default Missions;
