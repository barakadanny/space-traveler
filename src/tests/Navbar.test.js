import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import Header from "../components/Navbar";
import Missions from "../components/Missions/Missions";
import Profile from "../components/profile/Profile";
import Rockets from "../components/Rockets/Rockets";

import store from "../redux/configureStore";

describe("Header", () => {
  it("renders without crashing", () => {
    const header = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    expect(header).toMatchSnapshot();
  });
});

describe("Missions", () => {
  it("renders without crashing", () => {
    const mission = render(
      <Provider store={store}>
        <Router>
          <Missions />
        </Router>
      </Provider>
    );
    expect(mission).toMatchSnapshot();
  });
});

describe("Profile", () => {
  it("renders without crashing", () => {
    const profile = render(
      <Provider store={store}>
        <Router>
          <Profile />
        </Router>
      </Provider>
    );
    expect(profile).toMatchSnapshot();
  });
});

describe("Rockets", () => {
  it("renders without crashing", () => {
    const rocket = render(
      <Provider store={store}>
        <Router>
          <Rockets />
        </Router>
      </Provider>
    );
    expect(rocket).toMatchSnapshot();
  });
});
