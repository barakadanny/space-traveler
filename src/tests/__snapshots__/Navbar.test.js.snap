// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav
          class="navbar"
        >
          <div
            class="navbar-brand d-flex"
          >
            <img
              alt="logo"
              class="logo"
              src="planet.png"
            />
            <h1
              class="ms-3"
            >
              Space Traveler's Hub
            </h1>
          </div>
        </nav>
        <ul
          class="navbar"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
              style="text-decoration: underline; color: rgb(0, 0, 255);"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class=""
              href="/missions"
              style="text-decoration: none; color: rgb(18, 18, 18);"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class=""
              href="/profile"
              style="text-decoration: none; color: rgb(18, 18, 18);"
            >
              Profile
            </a>
          </li>
        </ul>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav
        class="navbar"
      >
        <div
          class="navbar-brand d-flex"
        >
          <img
            alt="logo"
            class="logo"
            src="planet.png"
          />
          <h1
            class="ms-3"
          >
            Space Traveler's Hub
          </h1>
        </div>
      </nav>
      <ul
        class="navbar"
      >
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
            style="text-decoration: underline; color: rgb(0, 0, 255);"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class=""
            href="/missions"
            style="text-decoration: none; color: rgb(18, 18, 18);"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class=""
            href="/profile"
            style="text-decoration: none; color: rgb(18, 18, 18);"
          >
            Profile
          </a>
        </li>
      </ul>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Missions renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mt-4"
      >
        <table
          class="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                Mission
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
              <th>
                  
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mt-4"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              Mission
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
            <th>
                
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Profile renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mt-4"
      >
        <div
          class="row"
        >
          <div
            class="col-md-4"
          >
            <h4>
              My Missions
            </h4>
            <ul
              class="subscriptions-container"
            />
          </div>
          <div
            class="col-md-4"
          >
            <h4>
              My Rockets
            </h4>
            <ul
              class="subscriptions-container"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mt-4"
    >
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <h4>
            My Missions
          </h4>
          <ul
            class="subscriptions-container"
          />
        </div>
        <div
          class="col-md-4"
        >
          <h4>
            My Rockets
          </h4>
          <ul
            class="subscriptions-container"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
